test/integration/vm_translator_test.rb

```diff
@@ -24,4 +24,10 @@ def test_integration_test_stack_test
     expected = File.read("test/fixtures/StackTest.asm")
     assert_equal(expected, assembly_code)
   end
+
+  def test_integration_test_basic_test
+    assembly_code = `bin/vm-translator examples/BasicTest.vm`
+    expected = File.read("test/fixtures/BasicTest.asm")
+    assert_equal(expected, assembly_code)
+  end
```

Comment:
We don’t need to fix it now, but note that this file is getting extremely repetitive: all of the tests are identical except for the `Foo.vm`/`Foo.asm` filenames. At some point we could clean this up with [metaprogramming](https://lpil.uk/blog/drying-up-minitest-tests-with-metaprogramming/) by writing a loop that calls `define_method` to dynamically define a test for each file we want to run through `bin/vm-translator`. (We could hardcode those filenames, but we could also determine them at runtime by [listing the contents](https://rubydoc.info/stdlib/core/Dir.glob) of the `examples/` directory.)

Reply:
Good idea! I created an issue so we remember this - https://github.com/wildmaples/jack-virtual-machine/issues/8